neuraxle/steps/data.py

```diff
@@ -0,0 +1,162 @@
+"""
+Data Steps
+====================================
+You can find here steps that take action on data.
+
+..
+    Copyright 2019, Neuraxio Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+"""
+import random
+from typing import Iterable
+
+from neuraxle.base import BaseStep, MetaStepMixin, NonFittableMixin, ExecutionContext
+from neuraxle.data_container import DataContainer
+from neuraxle.steps.output_handlers import InputAndOutputTransformerMixin
+
+
+class DataShuffler(NonFittableMixin, InputAndOutputTransformerMixin, BaseStep):
+    """
+    Data Shuffling step that shuffles data inputs, and expected_outputs at the same time.
+
+    .. code-block:: python
+
+        p = Pipeline([
+            TrainOnlyWrapper(DataShuffler(seed=42, increment_seed_after_each_fit=True)),
+            TrainOnlyWrapper(EpochRepeater(ForecastingPipeline(), epochs=EPOCHS))
+            TestOnlyWrapper(ForecastingPipeline())
```

Comment:
Wait, I think this is broken. There are two `ForecastingPipeline()` and the trained one won't be used in test mode...

Reply:
That is why I wanted to change the `EpochRepeater` from a wrapper to a simple base step: to allow wrapping it alone without shading it's wrapped step.

Reply:
I thought you only said that about the DataShuffler. okey I see I'll make the epoch repeater a base step as well.

Reply:
oh wait, but this is not possible !

Reply:
the parent step of the epoch repeater cannot do the loop, and it would be pretty ugly to dynamically wrap the step that needs to be repeated. resolving tell me if you had an idea on how to do this, but I cannot find one that makes sense @guillaume-chevalier resolving